cmd/podman/common/completion.go

```diff
@@ -1178,6 +1179,44 @@ func AutocompleteFormat(o interface{}) func(cmd *cobra.Command, args []string, t
 		if strings.HasPrefix("json", toComplete) {
 			return []string{"json"}, cobra.ShellCompDirectiveNoFileComp
 		}
+
+		// special(expensive) flow for "podman inspect"
+		if cmd != nil && cmd.Name() == "inspect" && cmd.Parent() == cmd.Root() {
+			if len(args) == 0 {
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			} else {
+				found := false
+				// container logic
+				if containers, _ := getContainers(cmd, args[0], completeDefault); len(containers) > 0 {
+					o = &define.InspectContainerData{}
+					found = true
+				}
+
+				// image logic
+				if images, _ := getImages(cmd, args[0]); len(images) > 0 && !found {
+					o = &inspect.ImageData{}
+					found = true
+				}
+
+				// volume logic
+				if volumes, _ := getVolumes(cmd, args[0]); len(volumes) > 0 && !found {
+					o = &define.InspectVolumeData{}
+					found = true
+				}
+
+				// pod logic
+				if pods, _ := getPods(cmd, args[0], completeDefault); len(pods) > 0 && !found {
+					o = &entities.PodInspectReport{}
+					found = true
+				}
+
+				// network logic
+				if networks, _ := getNetworks(cmd, args[0], completeDefault); len(networks) > 0 && !found {
+					o = &types.Network{}
+				}
```

Comment:
if you split this out to a new function then you do not need the extra `found` var and can just return early which would be much more readable

Reply:
will implement this logic to another function